### PR TITLE
Alloc ocf_data pages from pooled allocator

### PR DESCRIPTION
**Pull request by jfckm**
Signed-off-by: Jan Musial <jan.musial@intel.com>

### CODE REVIEW

File: module/bdev/ocf/ctx.c [hunk 1]
```diff
@@ -57,20 +58,24 @@ vbdev_ocf_ctx_data_alloc(uint32_t pages)
...
+	data = vbdev_ocf_data_alloc(pages);
+	if (!data)
 		return NULL;
+
+	for (i = 0; i < pages; i++) {
+		buf = env_allocator_new(ocf_data_allocator);
+
+		if (buf == NULL) {
+			return NULL;
+		}
```
Comment (robertbaldyga) [inline module/bdev/ocf/ctx.c]: Isn't a big ugly memleak?

**Review by robertbaldyga — CHANGES_REQUESTED** [review]

File: module/bdev/ocf/ctx.c (hunk 1 above)
Comment (jfckm) [reply module/bdev/ocf/ctx.c]: yes

File: module/bdev/ocf/ctx.c [hunk 2]
```diff
@@ -777,6 +783,11 @@ vbdev_ocf_ctx_init(void)
 		return ret;
 	}
 
+	ocf_data_allocator = env_allocator_create(PAGE_SIZE, "ocf_data");
```
Comment (mmichal10) [inline module/bdev/ocf/ctx.c:786]: Shouldn't the allocator be deinitialized on ocf ctx cleanup?

Comment (jfckm) [reply module/bdev/ocf/ctx.c:786]: fixed!

File: module/bdev/ocf/ctx.c [hunk 3]
```diff
@@ -86,12 +64,40 @@ vbdev_ocf_ctx_data_free(ctx_data_t *ctx_data)
...
+
+	data = vbdev_ocf_data_alloc(pages);
+	if (!data)
+		return NULL;
+
+	for (i = 0; i < pages; i++) {
+		buf = env_allocator_new(ocf_data_allocator);
+
+		if (buf == NULL) {
+			vbdev_ocf_ctx_data_free(data);
```
Comment (mmkayPL) [inline module/bdev/ocf/ctx.c:88]: Dealoc previous buffers as well

**Review by mmkayPL — APPROVED** [review]

**Review by arutk — APPROVED** [review]

**Pull request merged**